linkerd/main/src/main/scala/io/buoyant/linkerd/Main.scala

```diff
@@ -36,6 +36,7 @@ object Main extends App {
     args match {
       case Array(path) =>
         val config = loadLinker(path)
+        shutdownGracePeriod = config.admin.flatMap(_.shutdownGraceMs)
```

Comment:
why does this need to be a mutable `var` at all?

Reply:
We need to override a def `defaultCloseGracePeriod` with this value.  However, the value is loaded from the config file in the main method.  This means that when main is invoked and the config is read, this value must be stored somewhere so that it can be returned from `defaultCloseGracePeriod`.  Attempting to call `defaultCloseGracePeriod` before main has run will result in an NPE.